### DISCUSSION TITLE
When traversing with a Lua filter in top-down mode, exclude Metadata fields from external YAML file

### DISCUSSION THREAD

**Discussion by kopeckyf** [jgm/pandoc#9369] [Q&A] [answered]
With a Lua filter, I'm trying to traverse documents and an external YAML meta data file in `topdown` mode. My understanding is that metadata are accessible to Lua filters, but I thought that `topdown` traversal should make it possible to skip all meta data elements. 

The head of the AST of my documet looks like this:

```
Pandoc
  Meta
    { unMeta =
        fromList
          [ ( "abstract"
            , MetaInlines
                [ Str "Recent"
```

However, adding the function

```lua
function Meta (n)
  return n, false
end
```

takes no effect. In other words, the filter *is* applied to children of `Meta`. That same function works fine for `Cite` or `Link` objects, which are then skipped by the filter. 

Is it possible to exclude the entire `Meta` sub-tree from being processed by the filter?

**Comment by tarleb** [comment, marked as answer]
Indeed, looking at the code it seems that this is not implemented for Meta elements. Feel free to raise a bug.

My approach would be to place the filter in a table, and then use `walk` to apply the filter on the document body:

``` lua
local my_filter = {
  traverse = 'topdown',
  Str = function (s)
    -- ...
  end
}

function Pandoc(doc)
  doc.blocks = doc.blocks:walk(my_filter)
  return doc
end
```

**Reply by kopeckyf** [reply]
Nice idea, thank you! Will be opening an issue later

**Reply by kopeckyf** [reply]
Issue number for future reference: https://github.com/jgm/pandoc/issues/9381
